test-clients/src/main/java/com/hedera/services/bdd/suites/utils/precompile/FunctionResult.java

```diff
@@ -0,0 +1,68 @@
+package com.hedera.services.bdd.suites.utils.precompile;
+
+import com.esaulpaugh.headlong.abi.Tuple;
+import com.esaulpaugh.headlong.abi.TupleType;
+import java.math.BigInteger;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt256;
+
+public class FunctionResult {
+	private FunctionResult() {}
+
+	private static final TupleType mintReturnType = TupleType.parse("(int32,uint64,int64[])");
+	private static final TupleType burnReturnType = TupleType.parse("(int32,uint64)");
+
+	public static FunctionResult functionResult() {
+		return new FunctionResult();
+	}
+
+	public enum FunctionType {
+		MINT, BURN, TRANSFER, ASSOCIATE, DISASSOCIATE
+	}
+
+	private FunctionType functionType;
+	private TupleType tupleType;
+	private ResponseCodeEnum status;
+	private long totalSupply;
+	private long[] serialNumbers;
+
+	public FunctionResult forFunction(final FunctionType functionType) {
+		if (functionType == FunctionType.MINT) {
+			tupleType = mintReturnType;
+		} else {
+			tupleType = burnReturnType;
+		}
```

Comment:
Maybe modify this `else` to be an `else if (functionType == FunctionType.BURN)` in order to cover for the rest of the cases where FunctionType is neither MINT nor BURN.